pisa/stages/data/csv_loader.py

```diff
@@ -19,20 +20,71 @@ class csv_loader(Stage):  # pylint: disable=invalid-name
 
     Parameters
     ----------
-    events_file : csv file path
-    **kwargs
-        Passed to Stage
+
+    events_file : 
+        csv file path(s)
+
+    data_dict : str of a dict
+        Dictionary to specify what keys from the csv files to be loaded
+        under what name. Entries can be strings that point to the right
+        key in the csv file or lists of keys, and the data will be
+        stacked into a 2d array.
+        
+    output_names : sequence of str
+        Event categories to be recorded, needs to be a subset of names 
+        in `events_file`.
+
+    neutrinos : bool
+        Flag indicating whether data events represent neutrinos
+        In this case, special handling for e.g. nu/nubar, CC vs NC, ...
+
+    dis_idx : int
+        If there is no dis key but an interaction key, you need to specify
+        what interaction is dis
+        
+    scale_aeff : bool
+        Convert effective area from cm^2 to m^2 (PISA flux tables are stored in m^2)
+        if given in cm^2.
 
     """
     def __init__(
         self,
         events_file,
+        data_dict,
         output_names,
+        neutrinos=True,
+        dis_idx=None,
+        scale_aeff=False,
         **std_kwargs,
     ):
 
         # instantiation args that should not change
-        self.events_file = find_resource(events_file)
+        self.events_file = split(events_file)
+        for i, f in enumerate(self.events_file):
+            self.events_file[i] = find_resource(f)
+
+        if isinstance(data_dict, str):
+            self.data_dict = eval(data_dict)
+        elif isinstance(data_dict, dict):
+            self.data_dict = data_dict
+        else:
+            raise ValueError(
+                f"Unsupported type {type(data_dict)} for data_dict."
+            )
+
+        self.output_names = output_names
+        if len(self.output_names) != len(set(self.output_names)):
+            raise ValueError(
+                'Found duplicates in `output_names`, but each name must be'
+                ' unique.'
+            )
+
+        self.neutrinos = neutrinos
+        if not dis_idx is None:
```

Comment:
`if dis_idx is not None:` (https://peps.python.org/pep-0008/#programming-recommendations)